pyvista/ext/plot_directive.py

```diff
@@ -340,30 +342,41 @@ def render_figures(
     results = []
     ns = plot_context if context else {}
 
-    for i, code_piece in enumerate(code_pieces):
-        # generate the plot
-        _run_code(
-            doctest.script_from_examples(code_piece) if is_doctest else code_piece,
-            code_path,
-            ns,
-            function_name,
-        )
-
-        images = []
-        figures = pyvista.plotting._ALL_PLOTTERS
+    # Check for setup and teardown code for plots
+    code_setup = config.plot_setup
+    code_cleanup = config.plot_cleanup
 
-        for j, (_, plotter) in enumerate(figures.items()):
-            if hasattr(plotter, '_gif_filename'):
-                image_file = ImageFile(output_dir, f"{output_base}_{i:02d}_{j:02d}.gif")
-                shutil.move(plotter._gif_filename, image_file.filename)
-            else:
-                image_file = ImageFile(output_dir, f"{output_base}_{i:02d}_{j:02d}.png")
-                plotter.screenshot(image_file.filename)
-            images.append(image_file)
+    if code_setup:
+        _run_code(code_setup, code_path, ns, function_name)
```

Comment:
Since the surrounding code is tested, it's probably better to get this covered. I've added some trivial setup/cleanup code to the tinypages test where the rest is tested, see https://github.com/pyvista/pyvista/pull/2907/commits/de59e9fdeaa987b9c2051897fcd57e09012a4b57

Reply:
Thanks for the suggestion by the way, I forgot to look at coverage!